File: Java/src/Iniciante/Solucao_1012.java
import java.io.IOException;
import java.util.Scanner;


public class Main {
    public static double areaTriangulo(double base, double altura) {
        return (base * altura) / 2;
    }

    public static double areaRaioCirculo(double raio) {
        final double pi = 3.14159;
        return pi * Math.pow(raio, 2);
    }

    public static double areaTrapezio(double baseMaior, double baseMenor, double altura) {
        return ((baseMaior + baseMenor) * altura) / 2;

    }

    public static double areaQuadrado(double lado) {
        return lado * lado;
    }

    public static double areaRetagulo(double lado, double altura) {
        return lado * altura;
    }

    public static void main(String[] args) throws IOException {
        var input = new Scanner(System.in);

        double a = input.nextDouble();
        double b = input.nextDouble();
        double c = input.nextDouble();

        System.out.printf("TRIANGULO: %.3f%n", areaTriangulo(a,c));
        System.out.printf("CIRCULO: %.3f%n", areaRaioCirculo(c));
        System.out.printf("TRAPEZIO: %.3f%n", areaTrapezio(a,b,c));
        System.out.printf("QUADRADO: %.3f%n", areaQuadrado(b));
        System.out.printf("RETANGULO: %.3f%n", areaRetagulo(a,b));

        input.close();
    }
}

File: Python/Iniciante/Solucao_1012.py
def area_triangulo(base, altura):
    return (base * altura) / 2

def area_raio_circulo(raio):
    pi = 3.14159
    return pi * pow(raio, 2)

def area_trapezio(base_maior,base_menor,altura):
    return ((base_maior + base_menor)* altura) /2

def area_quadrado(lado):
    return lado * lado

def area_retangulo(lado, altura):
    return lado * altura
try:
    a ,b , c = map(float, input().split())
    #explicação da linha acima isso e utilizando para pega
    #vários elementos de uma vez em python funciona da seguinte forma
    #map e meio que um casting para um lote de elementos para não precisar
    #utilizar um for para fazer um casting de cada elemento
    #a função map tem como pârametro o tipo para a conversão e o iteravel
    #ja input().split() faz a seguinte coisa o input vai ler a String que o usario digitar
    #e o split() pega a strings tira todos os espaços e joga dentro de um vetor
    #exemplo ser chegar 10 90.90 20 isso vai chegar como string então "10 90.90 20"
    #ele pega e faz isso aqui ['10','90.90','20']
    #após isso o map transforma cada item em float então [10,90.90,20]
    #como temos a,b,c = [10,90.90,20] acontecer um desestruturação
    #ou seja a = 10 b = 90.90 c = 20

    print(f"TRIANGULO: {area_triangulo(a,c):.3f}\n"
          f"CIRCULO: {area_raio_circulo(c):.3f}\n"
          f"TRAPEZIO: {area_trapezio(a,b,c):.3f}\n"
          f"QUADRADO: {area_quadrado(b):.3f}\n"
          f"RETANGULO: {area_retangulo(a,b):.3f}")
except (ValueError, TypeError):
    exit()
